spiketoolkit/comparison/basecomparison.py

```diff
@@ -1,140 +1,79 @@
+from .comparisontools import (do_matching, do_score_labels, do_confusion_matrix)
 
 
 class BaseComparison:
     """
     Base class shared by SortingComparison and GroundTruthComparison
-    
     """
-    def __init__(self, sorting1, sorting2, sorting1_name=None, sorting2_name=None, delta_frames=10, min_accuracy=0.5,
-                n_jobs=1, verbose=False):
-        self._sorting1 = sorting1
-        self._sorting2 = sorting2
+    def __init__(self, sorting1, sorting2, sorting1_name=None, sorting2_name=None, delta_time=0.3, min_accuracy=0.5,
+                 n_jobs=1, verbose=False, sampling_frequency=None):
+        self.sorting1 = sorting1
+        self.sorting2 = sorting2
         if sorting1_name is None:
             sorting1_name = 'sorting 1'
         if sorting2_name is None:
             sorting2_name = 'sorting 2'
         self.sorting1_name = sorting1_name
         self.sorting2_name = sorting2_name
-        self._delta_frames = delta_frames
+        if self.sorting1.get_sampling_frequency() is not None:
+            assert self.sorting1.get_sampling_frequency() == self.sorting2.get_sampling_frequency(), \
+                "The two sorting extractors must have the same sampling frequency"
+            sampling_frequency = self.sorting1.get_sampling_frequency()
+        elif sampling_frequency is None:
+            raise Exception("SortingExtractors do not have sampling frequency information. Provide it with the "
+                            "'sampling_frequency' argument.")
+        self._delta_frames = int(delta_time * sampling_frequency)
         self._min_accuracy = min_accuracy
         self._n_jobs = n_jobs
         self.verbose = verbose
-        
-        # maching is done always
+
         self._do_matching()
-        
-        # make score label is done always
-        # Samuel EDIT : should do_score_label done on demand ?
-        # I think no.
         self._do_score_labels()
-        
+
         # confusion matrix is compute on demand
         self._confusion_matrix = None
 
-
-    def get_sorting1(self):
-        # Samuel EDIT : why not a direct attribute acees  with self.sorting1 ?
-        return self._sorting1
-
-    def get_sorting2(self):
-        # Samuel EDIT : why not a direct attribute acees  with self.sorting2 ?
-        return self._sorting2
-    
     def get_labels1(self, unit_id):
-        if unit_id in self._sorting1.get_unit_ids():
+        if unit_id in self.sorting1.get_unit_ids():
             return self._labels_st1[unit_id]
         else:
             raise Exception("Unit_id is not a valid unit")
 
     def get_labels2(self, unit_id):
-        if unit_id in self._sorting1.get_unit_ids():
+        if unit_id in self.sorting1.get_unit_ids():
             return self._labels_st1[unit_id]
         else:
             raise Exception("Unit_id is not a valid unit")
 
-
-
-    def plot_confusion_matrix(self, xlabel=None, ylabel=None, ax=None, count_text=True):
```

Comment:
No! Please keep it a bit more.
Please. i need a bit of time to see spike widget.
Please don't make unrelated to PR change.

Reply:
added back :)